compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -4958,7 +4958,8 @@ public void visit(BLangTypeConversionExpr conversionExpr) {
             return;
         }
         conversionExpr.typeNode = rewrite(conversionExpr.typeNode, env);
-        if (conversionExpr.type.tag == TypeTags.STRING && conversionExpr.expr.type.tag == TypeTags.XML_TEXT) {
+        if (conversionExpr.type.tag == TypeTags.STRING && (conversionExpr.expr.type.tag == TypeTags.XML_TEXT ||
+                types.isConversionExprXMLNeverUnion(conversionExpr))) {
```

Comment:
What are we trying to check here? Why do we need to go into the cast and check the expression?

Reply:
To  make sure we rewrite the xml for implicit casting